### DISCUSSION TITLE
Batdetect2 on pi

### DISCUSSION THREAD

**Discussion by BobSpong** [rdz-oss/BattyBirdNET-Pi#13] [Ideas]
Could this be made to run on a pi with Birdnet pi https://github.com/macaodha/batdetect2?

**Comment by rdz-oss** [comment]
Hi @BobSpong ,
if one puts in the software development work that can be done. It is a great detector for UK bats (and Mexico was it?). There are two things which make it a difficult choice, however:
- To train this for new species the data has to be manually prepared. For the BattyBirdNET-Analyzer you can just use recordings without much extra labelling work. So it can be extended more easily to other regions of the planet. e.g. Indonesia.
- The interface is different and would have to be adapted in a REST API. That however would be a one time effort and then done.
So if one values ease of extendability to arbitrary species the batdetect2 is not the first choice because of the manual labelling necessary.
The fastest way to get e.g. the US classifier going was via the BattyBirdNET-Analyzer approach.

But lets see, i am open to it if someone wants to help with that.
BR
Richard

**Reply by rdz-oss** [reply]
It looks like someone already went part of the way during a thesis work. Have only briefly looked at it but it might be a place to jump of if anyone is interested in developing a server.py offering batdetect2 behind it. The following repo has done part of that yet the results are not integrated into the database and UI (it looks, but I have not tried the system ), so not usable in this state as a standalone system. It appears that the focus was on storage and system load/performance. 
Cool work @NoxTheXelor !

https://github.com/NoxTheXelor/BatNET-Pi
